### PR TITLE
Basic project system

### PR DESCRIPTION
**Pull request by LPeter1997**
Added a basic project system, can be used to read up metadata references

### CODE REVIEW

File: src/Draco.ProjectSystem/Project.cs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+/// </summary>
+public sealed class Project
+{
+    /// <summary>
+    /// The project file.
+    /// </summary>
+    public FileInfo ProjectFile { get; }
+
+    /// <summary>
+    /// THe workspace the project belongs to.
```
Comment (Kuinox) [inline src/Draco.ProjectSystem/Project.cs:24]: typo

File: src/Draco.ProjectSystem/Project.cs [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    /// </summary>
+    public Workspace Workspace { get; }
+
+    /// <summary>
+    /// The target framework of the project.
+    /// </summary>
+    public string? TargetFramework
```
Comment (Kuinox) [inline src/Draco.ProjectSystem/Project.cs:31]: How is this related to the TargetFramework in a csproj ? 
There is a TargetFramework, and TargetFrameworks in csproj.

Comment (LPeter1997) [reply src/Draco.ProjectSystem/Project.cs:31]: It's the target FW we choose for the design time build

File: src/Draco.ProjectSystem/Project.cs (hunk 1 above)
Comment (LPeter1997) [reply src/Draco.ProjectSystem/Project.cs]: Fixed
